server/app/services/question/ActiveAndDraftQuestions.java

```diff
@@ -70,25 +69,35 @@ private ActiveAndDraftQuestions(Version active, Version draft, Version withDraft
                           Optional.ofNullable(draftNames.get(name)));
                     }));
 
-    draftVersionHasAnyEdits = draft.getPrograms().size() > 0 || draft.getQuestions().size() > 0;
+    draftVersionHasAnyEdits = draft.hasAnyChanges();
     referencingActiveProgramsByName = buildReferencingProgramsMap(active);
     referencingDraftProgramsByName =
         draftVersionHasAnyEdits ? buildReferencingProgramsMap(withDraftEdits) : ImmutableMap.of();
 
+    ImmutableSet<String> tombstonedQuestionNames =
+        ImmutableSet.copyOf(
+            Sets.union(
+                ImmutableSet.copyOf(draft.getTombstonedQuestionNames()),
+                ImmutableSet.copyOf(active.getTombstonedQuestionNames())));
     deletionStatusByName =
-        activeNames.keySet().stream()
+        versionedByName.keySet().stream()
             .collect(
                 ImmutableMap.toImmutableMap(
                     questionName -> questionName,
                     questionName -> {
-                      if (draft.getTombstonedQuestionNames().contains(questionName)) {
-                        return DeletionStatus.PENDING_DELETION;
-                      } else if (isNotDeletable(
-                          active, draft, activeNames, draftNames, questionName)) {
+                      final ImmutableMap<String, ImmutableSet<ProgramDefinition>>
+                          referencesToExamine =
```

Comment:
suggestion (non-blocking): i don't think the compiler requires this to be `final` so i recommend removing `final`

Reply:
Removed. I added it as final more as a hint to myself that we would only initialize the variable once. We don't really do that elsewhere so probably not worth starting here :D.